src/Adapter/Common/TagAwarePool.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of php-cache organization.
+ *
+ * (c) 2015-2016 Aaron Scherer <aequasi@gmail.com>, Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Cache\Adapter\Common;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\InvalidArgumentException;
+
+/**
+ * Interface for invalidating cached items using tags. This interface is a soon-to-be-PSR.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface TagAwarePool extends CacheItemPoolInterface
```

Comment:
We need to extend CacheItemPoolInterface inorder to make it type safe. 

Should we override the doc block for `getItem` and `getItems`?